source-code/core/module/src/resolveModules.ts

```diff
@@ -51,16 +49,27 @@ export const resolveModules: ResolvedModules = async (args) => {
 
 			// --- GET PLUGIN & LINT RULES ---
 			const plugins = inlangModule.default.plugins as Plugin[]
-			const lintRules = inlangModule.default.lintRules as ((...args: unknown[]) => LintRule)[]
+			const lintRules = inlangModule.default.lintRules as LintRule[]
 
-			const resolvedPlugins = resolvePlugins({
+			const resolvedPlugins = await resolvePlugins({
 				plugins,
 				pluginsInConfig,
 				config: args.config,
+				env: args.env,
 			})
 
-			if (resolvedPlugins.errors) {
-				result.errors.push(...resolvedPlugins.errors)
+			/**
+			 * -------------- BEGIN ADDING TO RESULT --------------
+			 */
+
+			result.data.resolvedPlugins = {
+				...result.data.resolvedPlugins,
+				...resolvedPlugins.data.plugins,
+			}
+
+			result.data.resolvedLintRules = {
+				...result.data.resolvedLintRules,
+				...lintRules,
```

Comment:
You can't spread here as the plugins and lint rules contain functions.

Reply:
(the function would be lost aka this code doesn't work)

Reply:
types & code are not ready, need to be improved - just fixed it for the build to pass

adding your comments to my list